sycl/source/device.cpp

```diff
@@ -150,6 +152,31 @@ __SYCL_EXPORT device device::get_info<info::device::parent_device>() const {
     return impl->template get_info<info::device::parent_device>();
 }
 
+template <>
+__SYCL_EXPORT std::vector<sycl::aspect>
+device::get_info<info::device::aspects>() const {
+  std::vector<sycl::aspect> DeviceAspects{
+#define __SYCL_ASPECT(ASPECT, ID) aspect::ASPECT,
+#include <sycl/info/aspects.def>
+#undef __SYCL_ASPECT
+  };
+
+  auto UnsupportedAspects = std::remove_if(
+      DeviceAspects.begin(), DeviceAspects.end(), [&](aspect Aspect) {
+        try {
+          return !impl->has(Aspect);
+        } catch (const runtime_error &ex) {
+          if (ex.get_cl_code() == PI_ERROR_INVALID_DEVICE)
```

Comment:
That's surprising for me that our `device_impl` can throw `INVALID_DEVICE` in `has` method? What's wrong with asking if a device has an aspect? How can device be invalid in that case?

Reply:
`device_impl::has` is implemented as a switch, it throws `PI_ERROR_INVALID_DEVICE` in the default case, so I thought it best to handle that one instance and rethrow others. Some testing I did threw this error so I added it, is it best to remove the try and let the exception be thrown normally and assume all aspects have an implemented code path?